test/integration.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
```

Comment:
ok, after 4 straight weeks of travel and late nights this will probably sound snippier than I intend and in the end it probably doesn't matter and I'll go with the will of the majority, but let me explain my thinking a little.... in a probably very disorganized way...

we're in the Cloud Native space - that means we're promoting containers, micro-services, etc...  It means when we talk about CN to our customers (or at least when I do) the idea of consistent reproducible environments are promoted.  We also push the idea of one task per container - e.g. we don't tell our customers to put their DB in the same container as their web server.  Yet, this PR and others that don't just allow people to "do their own thing" but go out of our way to accommodate it are promoting the exact opposite.  We're running a DB (etcd) and a server (apiserver) in the same container.  Why?  To allow people to have, in essence, their own build environment or to not use containers for whatever reason. This is not what we tell our customers about why containers are so great. And it smacks of "do as I say, not as I do", or "well, our needs are special".  That's just incorrect on both counts.

In the past I've seen cases where the build structure didn't work as well as it should. Rather than taking the time to fix it, people used what I would call "escape hatches" to avoid our build process.  That grinds me.  As an example, it was noticed that the compiles were slower than they needed to be because we didn't persist the .a file across builds. What happened? The build process was avoided rather than fixed.  grrrr  It didn't take a lot to fix it either - just the desire to actually do so.

To be clear, I don't really care that we have a large scbuildimage. Yes, I would prefer if we had one image for golang and one for glide, but for historical reasons we don't - and maybe one day we'll fix that.  But, what I really don't like is that we're spinning up one container and running two tasks (a DB and server) just so that people can avoid our build process or containers.  If there's something wrong, fix it. If its too slow, fix it. But I don't want people to use their own golang compiler or other tools. I want them to use the exact same ones as everyone else - and how do we tell our customers to get that? We say "use containers".   So, yes, if I had my way I'd kill the NO_DOCKER option.

I want us to eat our own CN dog food, and if it doesn't work then we should fix it. And by "it" I mean our build process, Docker, Kube, etc...

And no, a solution like I proposed doesn't require D-in-D.  

Sooooo looking forward to the weekend :-)